Branch-SDK/Branch-SDK/BNCError.m

```diff
@@ -6,10 +6,49 @@
 //  Copyright (c) 2014 Branch Metrics. All rights reserved.
 //
 
+
 #import "BNCError.h"
 
-NSString * const BNCErrorDomain = @"io.branch";
 
-@implementation BNCError
+NSString * const BNCErrorDomain = @"io.branch.error";
+
+
+#define _countof(arr) (sizeof(arr) / sizeof(arr[0]))
+
+
+NSError *_Nonnull BNCErrorWithCode(BNCErrorCode errorCode) {
+    return BNCErrorWithCodeAndReason(errorCode, nil);
+}
+
+
+NSError *_Nonnull BNCErrorWithCodeAndReason(BNCErrorCode errorCode, NSString* reason) {
+
+    NSString *const localizedDescriptions[] =
+        {
+         nil    //  0
```

Comment:
Can we just fill in localized descriptions even for the other errors. I know its outside the scope of the search Ads stuff but would be more maintainable even if the descriptions are not being used right now.